templates/kubernetes/terraform/modules/kubernetes/vpn.tf

```diff
@@ -0,0 +1,212 @@
+# Create VPN
+
+# generate VPN configuration
+locals {
+  namespace  = "vpn"
+
+  db_identifier          = "${var.project}-${var.environment}"
+  destination_subnets    = join(",", [for s in data.aws_subnet.my_db_subnet : s.cidr_block])
+  server_address	 = var.vpn_server_address
+  server_privatekey_name = "${var.project}-${var.environment}-vpn-wg-privatekey-${var.random_seed}"
+  client_peers           = join("\n", data.template_file.vpn_client_data_json.*.rendered)
+}
+
+## get destination database subnets
+data "aws_db_instance" "my_db" {
+  db_instance_identifier = local.db_identifier
+}
+data "aws_db_subnet_group" "my_db_subnetgroup" {
+  name = data.aws_db_instance.my_db.db_subnet_group
+}
+data "aws_subnet" "my_db_subnet" {
+  for_each = data.aws_db_subnet_group.my_db_subnetgroup.subnet_ids
+  id       = each.value
+}
+
+## TBD: get other destination subnets
+
+## get client public keys
+data "template_file" "vpn_client_data_json" {
+  template = file("${path.module}/files/wireguard-peer.tpl")
+  count    = length(var.vpn_client_publickeys)
+
+  vars = {
+    client_pub_key       = element(values(var.vpn_client_publickeys[count.index]), 0)
+    client_ip            = element(keys(var.vpn_client_publickeys[count.index]), 0)
+    persistent_keepalive = 25
+  }
+}
+
+data "aws_secretsmanager_secret" "vpn_private_key" {
+  name = local.server_privatekey_name
+}
+data "aws_secretsmanager_secret_version" "vpn_private_key" {
+  secret_id = data.aws_secretsmanager_secret.vpn_private_key.id
+}
+
+resource "kubernetes_namespace" "vpn_namespace" {
+  metadata {
+    name = local.namespace
+  }
+}
+
+resource "kubernetes_secret" "vpn_private_key" {
+  metadata {
+    name = "wg-secret"
+    namespace = local.namespace
+  }
+
+  data = {
+    privatekey = jsondecode(data.aws_secretsmanager_secret_version.vpn_private_key.secret_string)["key"]
+  }
+
+  type = "Opaque"
+}
+
+resource "kubernetes_config_map" "vpn_configmap" {
+  metadata {
+    name = "wg-configmap"
+    namespace = local.namespace
+  }
+
+  data = {
+    "wg0.conf" = "[Interface]\nAddress = ${local.server_address}\nListenPort = 51820\nPostUp = wg set wg0 private-key /etc/wireguard/privatekey && iptables -A FORWARD -s ${local.server_address} -d ${local.destination_subnets} -j ACCEPT && iptables -A FORWARD -s ${local.server_address} -j DROP && iptables -t nat -A POSTROUTING -s ${local.server_address} -o eth0 -j MASQUERADE\nPostDown = iptables -D FORWARD -s ${local.server_address} -d ${local.destination_subnets} -j ACCEPT && iptables -D FORWARD -s ${local.server_address} -j DROP && iptables -t nat -D POSTROUTING -s ${local.destination_subnets} -o eth0 -j MASQUERADE\n\n${local.client_peers}\n"
```

Comment:
How about using heredoc formatting here so it's not all on one line?

Reply:
done